### DISCUSSION TITLE
Major Overhaul v2.0.0

### DISCUSSION THREAD

**Discussion by NikolaiF90** [NikolaiF90/F90SimpleShopSystem#5] [Announcements]
### Version 2.0.0
* Complete overhaul of the whole system.
* Added Utils which contains dozens of scripts which might be useful for mission designers.
* Added guides. A detailed guide for setting a somewhat simple shop system.
* As simple as promised. Much more simple than expected.
* 10x easier to setup than previous version.
* Faster, optimized, bugless than previous version. 

**Full Changelog**:[v1.0.0>v2.0.0](https://github.com/NikolaiF90/F90SimpleShopSystem/blob/main/CHANGELOG.md)

<hr /><em>This discussion was created from the release <a href='https://github.com/NikolaiF90/F90SimpleShopSystem/releases/tag/v2.0.0'>Major Overhaul v2.0.0</a>.</em>

**Comment by Komaros** [comment]
Hi Nikolai
your script doesnt add addaction to object, when its on dedcated server, when i do it local it works.
Can you check ?

**Comment by NikolaiF90** [comment]
Im sorry. Currently its not optimized for multiplayer. Give me some time, I
will try to make it multiplayer compatible.

Btw, you can check my other latest project, Krasnaya Volchitsa. It has
somewhat similar shop system. And it's much more optimized for multiplayer.
You can extract the script and try if its suitable for your needs

**Reply by Komaros** [reply]
i will try, but you know im not programmer .... im to stupid to make something more comlicated.
im looking for shop system wchich can be implemented in multiplayer mission exactly in Liberation RX with multiple military units mods.
Your script seems to partally work on dedicated server, shop markers are created, there is just no addaction when player is at shop object.
Is there possible to paste manually in objects init addaction command somehow ?

**Comment by ethankil** [comment]
Was there any luck on making it multiplayer compatible from when you said you'd work on it 2 months back? I see it says it is compatible in the description but I didn't know if there were any changes.

**Comment by wspicher** [comment]
Hey I have a suggestion.  In a newer release, there should be a sell functionality for items, weapons, and vehicles.  For the items and weapons, you could have a container, placed by the shopkeeper/shop object, that serves as the item input. For vehicles, it could be proximity-based. Then, you could sell the items or vehicles for a slightly reduced price, compared to the new items.  I read through your code and it doesn't seem like this will be a heavy lift.  You have most of the helper functions in place.  I might be able to help add this through a PR if you're open to collaboration.  Just lmk.

**Comment by NikolaiF90** [comment]
Hey man. I heard your thoughts on this. Of course I'm open to
collaboration. Btw, which code do you refer to? Is it the original one or
the one integrated into the mission one? Cause the original one pretty much
abandoned. I have the remake on the mission

**Reply by wspicher** [reply]
Hey, I'm referring to the code last committed to about 7 months ago in [this folder](https://github.com/NikolaiF90/F90SimpleShopSystem/tree/main/F90SimpleShopSystem) of the repository.  I'm not sure if this is the original code you're referring to but this one looks like the most recently-contributed-to one.   Question for you, for editing the UI, did you use ARMA Dialog Creator or the in-game UI editor?  I'm just curious.
